src/vs/platform/storage/node/storageMainService.ts

```diff
@@ -179,4 +155,53 @@ export class StorageMainService extends Disposable implements IStorageMainServic
 		// Do it
 		return this.storage.close();
 	}
+
+	protected abstract createStorage(): IStorage;
+}
+
+export class StorageMainService extends AbstractStorageMainService implements IStorageMainService {
+
+	private static readonly STORAGE_NAME = 'state.vscdb';
+
+	constructor(
+		@ILogService private readonly logService: ILogService,
+		@IEnvironmentService private readonly environmentService: IEnvironmentService
+	) {
+		super();
+	}
+
+	private get storagePath(): string {
+		if (!!this.environmentService.extensionTestsLocationURI) {
+			return SQLiteStorageDatabase.IN_MEMORY_PATH; // no storage during extension tests!
+		}
+
+		return join(this.environmentService.globalStorageHome, StorageMainService.STORAGE_NAME);
+	}
+
+	protected createStorage(): IStorage {
+		return new Storage(new SQLiteStorageDatabase(this.storagePath, {
+			logging: this.createLogginOptions()
+		}));
+	}
+
+	private createLogginOptions(): ISQLiteStorageDatabaseLoggingOptions {
+		return {
+			logTrace: (this.logService.getLevel() === LogLevel.Trace) ? msg => this.logService.trace(msg) : undefined,
+			logError: error => this.logService.error(error)
+		};
+	}
+}
+
+export class SimpleStorageMainService extends AbstractStorageMainService implements IStorageMainService {
```

Comment:
How about we rename this and move it actually into `storageIpc.ts` to make clear this is a utility that can be used to have access to the global storage through IPC. Such as:
- in the ctor require the `MainProcessService`
- ctor would do the task of `getChannel('storage')` and using `GlobalStorageDatabaseChannelClient` for its DB
- we could name this `GlobalStorageService`

As an alternative: we could simply re-use `NativeStorageService` and make it optionally create the workspace storage by simply making the `IWorkspaceInitializationPayload` in `initialize` optional. That way the same storage service can be used from renderer and shared process with the only difference that the shared process cannot access any workspace scoped settings.

Reply:
I like the option of re-using `NativeStorageService`. This will also allow me to share the services depending storage service across renderer and shared. 

I think `NativeStorageService` is already handling no workspace storage and throws error when accessing. Only thing I have to do is make workspace payload optional during `initialise`.